docs/user/masternodes/setup-testnet.rst

```diff
@@ -925,19 +928,48 @@ Start the transaction filter stream::
 Envoy
 ^^^^^
 
-Envoy is a gRPC service proxy for cloud-native applications. Install
-Envoy as follows::
+Envoy is a gRPC service proxy for cloud-native applications. As it is
+responsible for TLS termination, you will need to obtain a TLS
+certificate for your public IP address before configuring Envoy. You can
+do this independently using a provider of your choice, or use ZeroSSL to
+obtain a free certificate. Visit https://app.zerossl.com/certificate/new
+and enter your public IP address, select HTTP File Upload as the
+verification method and then download the auth file. Place the contents
+of this file in the ``/var/www/html/.well-known/pki-validation/``
+directory with the original filename on your server and serve it using
+nginx as follows::
 
-  cd
-  curl -sL 'https://deb.dl.getenvoy.io/public/gpg.8115BA8E629CC074.key' | sudo gpg --dearmor -o /usr/share/keyrings/getenvoy-keyring.gpg
-  echo "deb [signed-by=/usr/share/keyrings/getenvoy-keyring.gpg] https://deb.dl.getenvoy.io/public/deb/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/getenvoy.list
-  sudo apt update
-  sudo apt install -y getenvoy-envoy
+  sudo mkdir -p /var/www/html/.well-known/pki-validation/
+  sudo nano /var/www/html/.well-known/pki-validation/<your_auth_file_name>.txt
+  sudo apt install -y nginx
+
+The auth file should be available at the following URL, for example::
+
+  http://3.27.105.157/.well-known/pki-validation/5F11ABBAE1FD926CB5BF8786A4E01492.txt
```

Comment:
> This seems like a mistake to me, that IP does not load, is it supposed to the IP of the host? and the hash, is it important?

This is intended as an example of what the finished location for the file should look like. I think maybe we can just delete this, because ZeroSSL provides a clickable URL in their interface for manual testing before they use the link to issue the certificate.